packages/backend/src/ee/services/McpService/McpService.ts

```diff
@@ -768,6 +789,120 @@ export class McpService extends BaseService {
+        // Exception: if user has no existing value for an attribute, admin can set it
+        for (const [key, overrideValues] of Object.entries(headerAttributes)) {
+            const existingValues = dbAttributes[key];
+
+            // If user has no existing values for this attribute, allow admin to set it
+            if (!existingValues || existingValues.length === 0) {
+                // eslint-disable-next-line no-continue
+                continue;
+            }
+
+            const hasWildcard = existingValues.includes('*');
+
+            if (!hasWildcard) {
+                const isSubset = overrideValues.every((v) =>
+                    existingValues.includes(v),
+                );
+                if (!isSubset) {
+                    throw new ForbiddenError(
+                        `Cannot expand access via header: values for '${key}' must be a subset of your existing permissions [${existingValues.join(
+                            ', ',
+                        )}]`,
+                    );
+                }
+            }
+        }
+    }
+
+    async getMergedUserAttributes(
+        context: McpProtocolContext,
+    ): Promise<UserAttributeValueMap> {
+        const { user, headerUserAttributes } = context.authInfo!.extra;
+        const { organizationUuid } = user;
+
+        if (!user || !organizationUuid) {
+            return {};
+        }
+
+        // Get database defaults
+        const dbAttributes =
+            await this.userAttributesModel.getAttributeValuesForOrgMember({
+                organizationUuid,
+                userUuid: user.userUuid,
+            });
+
+        // Validate header attributes if present (admin + narrowing check)
+        if (headerUserAttributes) {
+            McpService.validateHeaderUserAttributes(
+                user,
+                headerUserAttributes,
+                dbAttributes,
+            );
+        }
+
+        // Priority: header attributes > database defaults
+        return {
+            ...dbAttributes,
+            ...(headerUserAttributes || {}),
+        };
+    }
+
+    async getUserAttributeOverridesFromContext(
+        context: McpProtocolContext,
+    ): Promise<UserAttributeValueMap | undefined> {
+        const { user, headerUserAttributes } = context.authInfo!.extra;
+        const { organizationUuid } = user;
+
+        if (!user || !organizationUuid) {
+            return undefined;
+        }
+
+        if (!headerUserAttributes) {
+            return undefined;
+        }
+
+        // Validate header attributes (admin + narrowing check)
+        const dbAttributes =
+            await this.userAttributesModel.getAttributeValuesForOrgMember({
+                organizationUuid,
+                userUuid: user.userUuid,
+            });
+        McpService.validateHeaderUserAttributes(
+            user,
+            headerUserAttributes,
+            dbAttributes,
+        );
+
+        return headerUserAttributes;
+    }
```

Comment:
should these methods live in `userAttributesModel` or in utils near it? and MCP service just calls it